### ISSUE TITLE
bug: DaisyUI 5: `<html data-theme="light">` doesn't work when there's `prefersdark: true;`

### ISSUE THREAD

**Issue by lts20050703** [saadeghi/daisyui#3253]
### What version of daisyUI are you using?

5.0.0-alpha.29

### Which browsers are you seeing the problem on?

All browsers

### Reproduction URL

https://play.tailwindcss.com/pFadUDJMTg

### Describe your issue

`<html data-theme="light">` doesn't work when there's `prefersdark: true;`

app.css

```css
@import "tailwindcss";

@plugin "daisyui" {
	themes: light, dark;
}

@plugin "daisyui/theme" {
	name: "light";
	default: true;
}

@plugin "daisyui/theme" {
	name: "dark";
	prefersdark: true;
}
```

Strangely, `<html data-theme="dark">` still works when there's `prefersdark: true;`. 

I checked the generated CSS and found that `prefers-color-scheme: dark` is after `[data-theme="light"]`. This makes it impossible for `[data-theme="light"]` to override `prefers-color-scheme: dark`.

`data-theme="dark"` still works because it's after `prefers-color-scheme: dark`, therefore able to override `prefers-color-scheme: dark`.

TL;DR: Current generated CSS order:
- Root theme
- prefers-color-scheme: dark
- Other themes

Fixing this should be a simple reorder to
- prefers-color-scheme: dark
- Root theme
- Other themes

In other words, https://github.com/saadeghi/daisyui/blob/dfec9ea0f91552ff348b1e7dc0c26432ae2064d3/src/theming/functions.js#L214 needs to inject `prefers-color-scheme: dark` before injecting the root theme.

**Comment by saadeghi** [comment]
You choose the order.  
Change it to
```
@plugin "daisyui@5.0.0-alpha.29" {
	themes: dark, light;
}
```
and dark comes first in CSS.

Also, you can't change `<html>` tag in Tailwind Play. It will be ignored by Tailwind Play.

Thanks for testing daisyUI alpha 💚 and let me know if you have a question.

**Comment by lts20050703** [comment]
For anyone who got here, just want to say that swapping `themes: dark, light` to `themes: light, dark` isn't enough, you also have to move the dark style above the light style, something like this :)

```
@plugin "daisyui" {
	themes: dark, light;
}

@plugin "daisyui/theme" {
	name: "dark";
	prefersdark: true;
}

@plugin "daisyui/theme" {
	name: "light";
	default: true;
}
```

**Comment by hu3bi** [comment]
I have the same issue. Adding the proposed solution from @lts20050703 doesn't fix it.
If the device (root) theme is dark the theme switcher doesn't work.

**Comment by saadeghi** [comment]
@hu3bi can you share the CSS (order of themes)?

**Comment by hu3bi** [comment]
@saadeghi 
I tried:
`@plugin "daisyui" {
    themes: light --default, dark --prefersdark;
}`

and

`@plugin "daisyui" {
	themes: dark, light;
}

@plugin "daisyui/theme" {
	name: "dark";
	prefersdark: true;
}

@plugin "daisyui/theme" {
	name: "light";
	default: true;
}`


I found a fix for my issue. The problem is, that prefersdark is set the application switcher is no longer working when the user has a dark os theme.

`<input type="checkbox" class="theme-controller" value="dark"/>`

Changing the value to light for users who have a dark os theme is fixing it.


tldr: Fixed it by dynamically setting the value of the theme switcher to either light or dark.

**Comment by saadeghi** [comment]
@hu3bi I can't see any issues: https://play.tailwindcss.com/wP5gmVeumt

**Comment by hu3bi** [comment]
@saadeghi this doesn't work for me: https://play.tailwindcss.com/LGkYlB6pAN
What should be the behavior here: If the user has a dark os theme, the theme switcher should still work and switch to a light theme when pressed.


This works: https://play.tailwindcss.com/y2tEVYCu0V

If needed, I can record a video.

**Comment by saadeghi** [comment]
@hu3bi This is expected behavior.
`<input type="checkbox" class="theme-controller" value="dark" />` doesn't switch between `light` and `dark`. It switches between "current theme" and the theme in its value (`dark` in this example)

When browser is in dark mode, it switches between dark and dark.  
To make it work, remove `--prefersdark`, so the default theme becomes `light`,  
now `<input type="checkbox" class="theme-controller" value="dark" />` switches correctly between default (light) and its value (dark):  
https://play.tailwindcss.com/oiQxsDkv3K?file=css

Let me know if you have a question.

**Comment by hu3bi** [comment]
@saadeghi Thanks for the clarification. My apologies.
